java/hcx-integrator-sdk/src/main/java/io/hcxprotocol/impl/HCXOutgoingRequest.java

```diff
@@ -149,19 +151,30 @@ public boolean encryptPayload(Map<String,Object> headers, String fhirPayload, Ma
 
     // Exception is handled in processFunction method
     @Override
-    public boolean initializeHCXCall(String jwePayload, Operations operation, Map<String,Object> response, Config config) throws Exception {
-        Map<String,String> headers = new HashMap<>();
-        headers.put(Constants.AUTHORIZATION, "Bearer " + Utils.generateToken(config.getString(Constants.USERNAME), config.getString(Constants.PASSWORD), config.getString(Constants.AUTH_BASE_PATH)));
-        HttpResponse hcxResponse = HttpUtils.post(config.getString(Constants.PROTOCOL_BASE_PATH) + operation.getOperation(), headers, jwePayload);
-        response.put(Constants.RESPONSE_OBJ, JSONUtils.deserialize(hcxResponse.getBody(), Map.class));
-        int status = hcxResponse.getStatus();
-        boolean result = false;
-        if(status == 202){
-            result = true;
-            logger.info("Processing outgoing request has completed ::  response: {}", response.get(Constants.RESPONSE_OBJ));
-        } else {
-            logger.error("Error while processing the outgoing request :: status: {} :: response: {}", status, response.get(Constants.RESPONSE_OBJ));
-        }
+    public boolean initializeHCXCall(String jwePayload, Operations operation, Map<String, Object> response, Config config) throws Exception {
+        boolean result;
+        result = Utils.initializeHCXCall(jwePayload, operation, response, config);
         return result;
     }
+
+    @Override
+    public boolean sendNotification(String topicCode, String recipientType, List<String> recipients, String message, Map<String, String> templateParams, String correlationId, Map<String, Object> output, Config config) throws Exception {
+        try {
+            boolean result;
+            NotificationRequest notificationRequest = new NotificationRequest(topicCode, message, templateParams, recipientType, recipients, correlationId, config);
+            NotificationService.validateNotificationRequest(notificationRequest);
+            Map<String, Object> requestBody = NotificationService.createNotificationRequest(notificationRequest, output, message);
+            Map<String, Object> response = new HashMap<>();
+            result = initializeHCXCall(JSONUtils.serialize(requestBody), Operations.NOTIFICATION_NOTIFY, response, config);
+            output.putAll(response);
```

Comment:
why creating new response map and pass to initializeHCXCall? pass output map to the method.